src/web_scraper/web_scraper.py

```diff
@@ -0,0 +1,125 @@
+        bill_no = re.findall(bill_no_regex, bill.title)[0].split(' ')[-1]
+        
+        printer_no_regex="Printer's Number \d*"
+        printer_no = re.findall(printer_no_regex, bill.title)[0].split(' ')[-1]
+
+        #OLD CODE that doesn't work
+        #title_elements = bill.title.split(' ') #grabbing bill and printer numbers from the title (not explicitly given in RSS feed)
+        #bill_no = "".join(char for char in title_elements[2] if char.isdigit()) #ensuring that only the actual digits of the bill number are extracted
+        #printer_no = "".join(char for char in title_elements[5] if char.isdigit()) #ensuring that only the actual digits of the printer number are extracted
+
+        #change if bill["parss_enacted"] == "YES" and title_elements[1] == "Bill": #make sure the bill was enacted and it's a bill instead of a resolution        
+        bill_existing_records = supabase.table('Revisions').select("*", count="exact").eq("bill_id", bill_no).eq("printer_no", printer_no).eq("legislative_body", "senate").execute()
+        if bill_existing_records.count == 0: #if the bill hasn't already been recorded in the supabase table, then post it (so you don't upload duplicates)
+            response = supabase.table('Revisions').insert( #why is there no python docs for upsert?
+                {
+                    # "created_at": metadata["created_at"], #supabase already defaults to now(), so this is kind of unnecessary for now
+                    "bill_id": bill_no,
+                    "printer_no": printer_no, 
+                    "full_text_link": bill["link"], 
+                    "publication_date": bill["published"], 
+                    "legislative_body": "senate"
+                    # "active_summary_id": metadata["active_summary_id"], #leaving out foreign keys for now
+                }).execute()
+            #print(response)
+    return
+
+def extract_and_upload_house_bill_metadata():
+    house_rss = feedparser.parse(house_rss_feed)
+    supabase: Client = create_client(sb_api_url, sb_api_key)
+    for bill in house_rss.entries:
+        #print("checking bill ", bill.title)
+        if bill is None: continue
+
+        bill_no_regex = '(House Bill \d*|House Resolution \d*)'
+        bill_no = re.findall(bill_no_regex, bill.title)[0].split(' ')[-1]
+
+        printer_no_regex="Printer's Number \d*"
+        printer_no = re.findall(printer_no_regex, bill.title)[0].split(' ')[-1]
+
+        #old code that doesn't work
+        #title_elements = bill.title.split(' ')#grabbing bill and printer numbers from the title (not explicitly given in RSS feed)
+        #bill_no = "".join(char for char in title_elements[2] if char.isdigit()) #ensuring that only the actual digits of the bill number are extracted
+        #printer_no = "".join(char for char in title_elements[5] if char.isdigit()) #ensuring that only the actual digits of the printer number are extracted
+
```

Comment:
As in the other case, let's remove commented out old code, since we can just consult it in version control later 🍔